### PR TITLE
Core/admin refactoring

### PR DESCRIPTION
**Pull request by soupette**
Translations will be loaded differently in another user story

Signed-off-by: soupette <cyril.lpz@gmail.com>

### What does it do?

- [x] Remove immutable.js in CM core/content-manager/pages/ListSettingsView/reducer.js
- [x] Remove immutable.js in CM core/content-manager/pages/EditSettingsView/reducer.js
- [x] Add more tests in CM
- [x] Refactor core/admin/pages/Admin
- [x] Rename containers to pages in CM
- [x] Rename containers to pages in upload plugin
- [x] Rename containers to pages in CTB
- [x] Rename containers to pages in Documentation
- [x] Rename containers to pages in i18n plugin
- [x] Rename containers to pages in users-permissions
- [x] Remove immutable.js in upload plugin
- [x] Create a common test utils package for the admin 

### Why is it needed?

TODO

### How to test it?

TODO

### CODE REVIEW

File: packages/core/admin/admin/src/pages/Admin/index.js [hunk 1]
```diff
@@ -4,281 +4,121 @@
...
+  import(/* webpackChunkName: "Admin_pluginDispatcher" */ '../PluginDispatcher')
+);
+const ProfilePage = lazy(() =>
+  import(/* webpackChunkName: "Admin_profilePage" */ '../ProfilePage')
+);
+const SettingsPage = lazy(() =>
+  import(/* webpackChunkName: "Admin_settingsPage" */ '../SettingsPage')
+);
+// These are internal plugins
+const CM = lazy(() =>
```
Comment (soupette) [inline packages/core/admin/admin/src/pages/Admin/index.js:44]: @alexandrebodin

File: packages/core/admin/webpack.config.js [hunk 2]
```diff
@@ -91,7 +91,8 @@ module.exports = ({
       rules: [
         {
           test: /\.m?js$/,
-          exclude: /node_modules/,
+          // TODO remove when plugins are built separately
+          exclude: /node_modules\/(?!(@strapi\/plugin-content-manager|@strapi\/plugin-content-type-builder|@strapi\/plugin-upload)\/).*/,
```
Comment (soupette) [inline packages/core/admin/webpack.config.js:95]: @alexandrebodin this is temporary until we build the plugins separately, if you now a better regex I'm in.
Currently this is what we prevent us to do some lazy loading with the not required plugins, the strapi-plugin-* (from the community) and the local ones.

Comment (alexandrebodin) [reply packages/core/admin/webpack.config.js:95]: I think like this should be ignored automatically by webpack if the package has the right properties in its package.json 🤔

Comment (alexandrebodin) [reply packages/core/admin/webpack.config.js:95]: that's good enough for a temp regex

File: packages/core/admin/admin/src/pages/Admin/index.js (hunk 1 above)
Comment (alexandrebodin) [reply packages/core/admin/admin/src/pages/Admin/index.js:44]: I see. I'm still not convinced by this. feels like the plugin should do it themselves

```js
import CTB from '@strapi/ctb/admin';


```

and in CTB
`index.js`
```js
export default () => {

  return {
    main: lazy(() => import('./Main'))
  }
}
```

**Review by HichamELBSI — APPROVED** [review]
LGTM

File: packages/core/admin/admin/src/pages/Admin/index.js (hunk 1 above)
Comment (soupette) [reply packages/core/admin/admin/src/pages/Admin/index.js:44]: the CTB, CM & upload are required ones I think it is easier if we load them directly in the admin package, however it does not prevent us from keeping the actual way of loading the plugin and adding lazy.

**Pull request merged**